### PR TITLE
Shader manager

### LINKED ISSUE #260: Changes to shader API in VTK 9
When running FURY shader code with VTK 9, I now see warnings like this:

```
vtkOpenGLPolyDataMapper:227   WARN| 
vtkOpenGLPolyDataMapper::AddShaderReplacement was deprecated for VTK 9.0 and will be removed in a future version.  
Use vtkOpenGLShaderProperty::AddShaderReplacement instead.
```

The new implementation comes from [this commit](https://github.com/Kitware/VTK/commit/1b005a294d34f5628dc029fa943047cff9629f22), where the new classes and some examples of their use can be found.

We probably don't need to do anything until:

- FURY moves to a requirement of `vtk>=9.0.0` or
- VTK removes the feature instead of just deprecating it

### PR DESCRIPTION
**Pull request by skoudoro**
Beginning of adding utility functions to standardize and help the shader management. Still some work to do:

- [x] Add tests
- [x] Add callback
- [x] Add attributes

Feel free to comment and add more ideas. 

fix #260

### CODE REVIEW

**Comment by lenixlobo** [comment]
Hi! 
This is very easy to use . 

In documentation, I think we should also explain the shaders blocks in short
As a new user, it can be quite confusing why the blocks exist and vtk doesn’t have much documentation on the same.

**Comment by lenixlobo** [comment]
Hi! 
This is very easy to use . 

In documentation, I think we should also explain the shaders blocks in short
As a new user, it can be quite confusing why the blocks exist and vtk doesn’t have much documentation on the same.

File: fury/shaders/base.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        raise ValueError(msg)
+
+    block_dec = block + "::Dec"
+    block_impl = block + "::Impl"
+
+    if keep_default:
+        decl_code = block_dec + "\n" + decl_code
+        impl_code = block_impl + "\n" + impl_code
+
+    if debug:
```
Comment (guaje) [inline fury/shaders/base.py:79]: Nice one! This will be extremely useful.

File: fury/shaders/base.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    Parameters
+    ----------
+    actor : vtkActor
+        Rendered Object
+    arr : ndarray
+        array to link to vertices
+    arr_name : str
+        data array name
+    attr_name : str
```
Comment (guaje) [inline fury/shaders/base.py:122]: Could we just keep one of the 2 params names in the signature of the function and if we need to differentiate between the 2 names do it by appending a fixed prefix to one of the 2 strings. My suggestion would be to leave the users to define the attribute they have to retrieve in the vertex shader and leave this function to manage the name of the vtk array.

File: fury/shaders/base.py (hunk 1 above)
Comment (Garyfallidis) [reply fury/shaders/base.py:79]: ;)

File: fury/shaders/base.py (hunk 2 above)
Comment (skoudoro) [reply fury/shaders/base.py:155]: 👍

**Comment by skoudoro** [comment]
PR ready for review @guaje @Garyfallidis @lenixlobo @dmreagan.

**Comment by Garyfallidis** [comment]
I suggest not using ``add_`` in these functions. If you have add then you need to have remove too.
What about using ``shader_to_actor`` and ``attribute_to_actor``.

**Comment by Garyfallidis** [comment]
Merging. On another PR we will have to update the experimental shaders too.

**Pull request merged**